lib/logstash/outputs/datadog_metrics.rb

```diff
@@ -88,24 +88,15 @@ def receive(event)
     buffer_receive(dd_metrics)
   end # def receive
 
-  public
+  # public
   def flush(events, final=false)
     dd_series = Hash.new
-    dd_series['series'] = []
-
-    events.each do |event|
-      begin
-        dd_series['series'] << event
-      rescue
-        @logger.warn("Error adding event to series!", :exception => e)
-        next
-      end
-    end
+    dd_series['series'] = Array(events).flatten
```

Comment:
There is almost not cost incurred here if events is an array already.  Perhaps flatten is overkill. :-)